src/enftun.c

```diff
@@ -225,15 +232,24 @@ enftun_connect(struct enftun_context* ctx)
             goto out;
     }
 
+#ifndef USE_PSOCK
     if ((rc = enftun_conn_state_prepare(&ctx->conn_state, &ctx->loop,
                                         trigger_reconnect, (void*) ctx,
                                         ctx->config.fwmark)) < 0)
+    {
         goto out;
+    }
 
     if ((rc = enftun_tls_connect(&ctx->tls, ctx->config.fwmark,
                                  ctx->config.remote_hosts,
                                  ctx->config.remote_port)) < 0)
         goto close_conn_state;
+#else
+    if ((rc = enftun_tls_proxy_connect(&ctx->tls, ctx->config.remote_hosts,
```

Comment:
Specializing the TLS connect function to be proxy-aware doesn't seem right, since the difference is just in the underlying tcp socket.

The XTT code will also need to work with both regular and proxy TCP sockets.

Let's think about ways to encapsulate the TCP/psock options into one interface that both the TLS and the XTT code can use.